src/threadpool.c

```diff
@@ -161,7 +160,11 @@ static void init_threads(void) {
     if (uv_thread_create(threads + i, worker, NULL))
       abort();
 
-  initialized = 1;
+  do {
+    uv_mutex_lock(&mutex);
+    spin = (idle_threads < nthreads);
+    uv_mutex_unlock(&mutex);
+  } while (spin);
```

Comment:
Spinning like this is admittedly not very nice but it only happens once and only for microseconds.

Reply:
Isn't it possible for one of the threads to pickup a work item and decrement `idle_threads`?

Reply:
No.  init_threads() runs exclusively, it's called from the `uv_once()` in `uv__work_submit()`.

Reply:
Right. LGTM then!